src/nitypes/waveform/__init__.py

```diff
@@ -53,6 +53,93 @@
 >>> wfm.scaled_data
 array([2.5, 4.5, 6.5])
 
+Timing Information
+------------------
+
+Analog waveforms include timing information, such as the start time and sample interval, to support
+analyzing and visualizing the data.
+
+You can specify timing information by constructing a :any:`Timing` object and passing it to the
+waveform constructor or factory method:
+
+>>> import datetime as dt
+>>> wfm = AnalogWaveform(timing=Timing.create_with_regular_interval(
+...     dt.timedelta(seconds=1e-3), dt.datetime(2024, 12, 31, 23, 59, 59, tzinfo=dt.timezone.utc)
+... ))
+>>> wfm.timing  # doctest: +NORMALIZE_WHITESPACE
+nitypes.waveform.Timing(nitypes.waveform.SampleIntervalMode.REGULAR,
+    timestamp=datetime.datetime(2024, 12, 31, 23, 59, 59, tzinfo=datetime.timezone.utc),
+    sample_interval=datetime.timedelta(microseconds=1000))
+
+You can query the waveform's timing information using the :any:`Timing` object's properties:
+
+>>> wfm.timing.start_time
+datetime.datetime(2024, 12, 31, 23, 59, 59, tzinfo=datetime.timezone.utc)
+>>> wfm.timing.sample_interval
+datetime.timedelta(microseconds=1000)
+
+Timing objects are immutable, so you cannot directly set their properties:
+
+>>> wfm.timing.sample_interval = dt.timedelta(seconds=10e-3)
+Traceback (most recent call last):
+...
+AttributeError: property 'sample_interval' of 'Timing' object has no setter
+
+Instead, if you want to modify the timing information for an existing waveform, you can create a new
+timing object and set the :any:`AnalogWaveform.timing` property:
```

Comment:
```
/home/runner/work/nitypes-python/nitypes-python/docs/autoapi/nitypes/waveform/index.rst:88: WARNING: 'any' reference target not found: AnalogWaveform.timing [ref.any]
/home/runner/work/nitypes-python/nitypes-python/docs/autoapi/nitypes/waveform/index.rst:116: WARNING: 'any' reference target not found: AnalogWaveform.timing [ref.any]
```

Reply:
The property is defined on the base class `NumericWaveform`, not the derived class `AnalogWaveform`